crates/store/src/db/sql.rs

```diff
@@ -307,6 +307,51 @@ pub fn select_nullifiers_by_block_range(
     Ok(result)
 }
 
+/// Select nullifiers created that match the `nullifier_prefixes` filter using the given
+/// [Connection].
+///
+/// Each value of the `nullifier_prefixes` is only the `prefix_len` most significant bits
+/// of the nullifier of interest to the client. This hides the details of the specific
+/// nullifier being requested.
+///
+/// # Returns
+///
+/// A vector of [NullifierInfo] with the nullifiers and the block height at which they were
+/// created, or an error.
+pub fn select_nullifiers_by_prefix(
+    conn: &mut Connection,
+    _prefix_len: u32,
+    nullifier_prefixes: &[u32],
+) -> Result<Vec<NullifierInfo>> {
+    let nullifier_prefixes: Vec<Value> =
+        nullifier_prefixes.iter().copied().map(u32_to_value).collect();
+
+    let mut stmt = conn.prepare(
```

Comment:
@bobbinth separate note: we should consider using [prepare_cached](https://docs.rs/rusqlite/latest/rusqlite/struct.Connection.html#method.prepare_cached). Parsing a sql statement is actually a non-trivial cost.

I'm not suggesting this PR changes this since `prepare` is used in all the other queries as well.

Reply:
Agreed! Let's create an issue for this.

Reply:
#423